### PR TITLE
Add limit to 256 characters when creating a quiz

### LINKED ISSUE #75: Creating Quizzes does not work for longer inputs but gives no error
### Bootstrap Academy username
CptK

### Contact Details
marcus.kornmann@stud.tu-darmstadt.de

### What browsers are you seeing the problem on?
Chrome

### What happened?

I have noticed that creating quizzes fails if the input is too long (related to #73 ). For now, if you click on create it does not work but nothing happens. Only if you open the console you see some kind of error:
image

The user should get an error message telling the problem.

### PR DESCRIPTION
**Pull request by Lauritz-Tieste**
## Description
I added a limit to 256 characters when creating a quiz, because the backend cannot handle more.

fixes https://github.com/Bootstrap-Academy/Bootstrap-Academy/issues/75

My Bootstrap Academy username: Lauritz

### CODE REVIEW

File: components/form/Quiz.vue [hunk 1]
```diff
@@ -336,6 +336,11 @@ export default defineComponent({
         }
         if (hasDuplicates(options.value))
           return openSnackbar("error", "Error.OptionsCannotBeSame");
+        for (let i = 0; i < options.value.length; i++) {
+          if (options.value[i].answer.length > 255) {
+            return openSnackbar("error", "Error.OptionCannotBeMoreThen256");
```
Comment (Defelo) [inline components/form/Quiz.vue]:
```suggestion
            return openSnackbar("error", "Error.OptionCannotBeMoreThan256");
```

Also the translation key seems to be missing

**Review by Defelo — CHANGES_REQUESTED** [review]
The length of the question should also be checked

**Review by Lauritz-Tieste — COMMENTED** [review]
After all the back and forth with the commits, it should work now

File: components/form/Quiz.vue [hunk 2]
```diff
@@ -336,6 +336,14 @@ export default defineComponent({
...
         if (hasDuplicates(options.value))
           return openSnackbar("error", "Error.OptionsCannotBeSame");
+        for (let i = 0; i < options.value.length; i++) {
+          if (options.value[i].answer.length > 255) {
+            return openSnackbar("error", t("Error.CannotHaveMoreCharacters", { input: t("Inputs.AnswerOption"), max: 256 }));
+          }
+        }
+        if (form.question.value.length > 255) {
+          return openSnackbar("error", t("Error.CannotHaveMoreCharacters", { input: t("Inputs.Question"), max: 256 }));
+        }
```
Comment (Defelo) [inline components/form/Quiz.vue]: The maximum length of the question is actually 4096, see https://api.test.bootstrap.academy/challenges/redoc#tag/MultipleChoice/paths/~1tasks~1%7Btask_id%7D~1multiple_choice/post

```suggestion
        if (form.question.value.length > 4096) {
          return openSnackbar("error", t("Error.CannotHaveMoreCharacters", { input: t("Inputs.Question"), max: 4096 }));
        }
```

File: components/form/Quiz.vue [hunk 3]
```diff
@@ -336,6 +336,14 @@ export default defineComponent({
         }
         if (hasDuplicates(options.value))
           return openSnackbar("error", "Error.OptionsCannotBeSame");
+        for (let i = 0; i < options.value.length; i++) {
+          if (options.value[i].answer.length > 255) {
```
Comment (Defelo) [inline components/form/Quiz.vue]: Why 255? The maximum length of each option is 256 (inclusive), so 256 characters are still allowed.
```suggestion
          if (options.value[i].answer.length > 256) {
```

**Review by Defelo — CHANGES_REQUESTED** [review]

**Review by Defelo — APPROVED** [review]

**Pull request merged**